sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/datastore/DataStoreV1SchemaIOProviderTest.java

```diff
@@ -0,0 +1,99 @@
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+
+import org.apache.beam.sdk.io.gcp.datastore.DataStoreV1SchemaIOProvider.DataStoreV1SchemaIO;
+import org.apache.beam.sdk.schemas.Schema;
+import org.apache.beam.sdk.schemas.io.SchemaIO;
+import org.apache.beam.sdk.values.Row;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class DataStoreV1SchemaIOProviderTest {
+  static final String DEFAULT_KEY_FIELD = "__key__";
+  public static final String KEY_FIELD_PROPERTY = "keyField";
+  private DataStoreV1SchemaIOProvider provider = new DataStoreV1SchemaIOProvider();
+
+  @Test
+  public void testGetTableType() {
+    assertEquals("datastoreV1", provider.identifier());
+  }
+
+  @Test
+  public void testBuildBeamSqlTable() {
+    final String location = "projectId/batch_kind";
+    Row configuration = Row.withSchema(generateRowSchema()).addValue(null).build();
+    SchemaIO schemaIO =
+        (new DataStoreV1SchemaIOProvider()).from(location, configuration, generateDataSchema());
+
+    assertNotNull(schemaIO);
+    assertTrue(schemaIO instanceof DataStoreV1SchemaIO);
+
+    DataStoreV1SchemaIO dataStoreV1SchemaIO = (DataStoreV1SchemaIO) schemaIO;
+    assertEquals("projectId", dataStoreV1SchemaIO.projectId);
+    assertEquals("batch_kind", dataStoreV1SchemaIO.kind);
+    assertEquals(DEFAULT_KEY_FIELD, dataStoreV1SchemaIO.keyField);
+  }
+
+  @Test
+  public void testTableProperty() {
+    final String location = "projectId/batch_kind";
+
+    Row configuration = Row.withSchema(generateRowSchema()).addValue("field_name").build();
+    SchemaIO schemaIO =
+        (new DataStoreV1SchemaIOProvider()).from(location, configuration, generateDataSchema());
+
+    assertNotNull(schemaIO);
+    assertTrue(schemaIO instanceof DataStoreV1SchemaIO);
+
+    DataStoreV1SchemaIO dataStoreV1SchemaIO = (DataStoreV1SchemaIO) schemaIO;
+    assertEquals("projectId", dataStoreV1SchemaIO.projectId);
+    assertEquals("batch_kind", dataStoreV1SchemaIO.kind);
+    assertEquals("field_name", dataStoreV1SchemaIO.keyField);
+  }
+
+  @Test
+  public void testTableProperty_nullValue_throwsException() {
+    final String location = "projectId/batch_kind";
+    Row configuration = Row.withSchema(generateRowSchema()).addValue("").build();
+
+    assertThrows(
+        IllegalArgumentException.class,
+        () ->
+            (new DataStoreV1SchemaIOProvider())
+                .from(location, configuration, generateDataSchema()));
+  }
+
+  private static Schema generateDataSchema() {
+    return Schema.builder()
+        .addNullableField("id", Schema.FieldType.INT32)
+        .addNullableField("name", Schema.FieldType.STRING)
+        .build();
+  }
+
+  private Schema generateRowSchema() {
+    return Schema.builder().addNullableField(KEY_FIELD_PROPERTY, Schema.FieldType.STRING).build();
+  }
```

Comment:
I think it would be better to get rid of this function and call `provider.configurationSchema()` instead.

Reply:
Done.